web/modules/weather_cms/src/Timezones.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Drupal\weather_cms;
+
+class Timezones
+{
+    /**
+     * Abbreviations of DateTimeZone instances
+     * that cover the US and territories, mapped
+     * to longer names we have take from NIST at
+     * time.gov
+     */
+    private static $ABBREVS_TO_LONGNAME = [
+        "UTC" => "Universal Coordinated Time (UTC)",
+        "AKDT" => "Alaska Daylight Time",
+        "AKST" => "Alaska Standard Time",
+        "HDT" => "Hawaii–Aleutian Daylight Time",
+        "HST" => "Hawaii–Aleutian Standard Time",
+        "MDT" => "Mountain Daylight Time",
+        "MST" => "Mountain Standard Time",
+        "CDT" => "Central Daylight Time",
+        "CST" => "Central Standard Time",
+        "EDT" => "Eastern Daylight Time",
+        "EST" => "Eastern Standard Time",
+        "PDT" => "Pacific Daylight Time",
+        "PST" => "Pacific Standard Time",
+        "ChST" => "Chamorro Standard Time (Guam)",
+        "SST" => "Samoa Standard Time",
+    ];
+
+    /**
+     * Drupal DateTimeZone id names that are each
+     * an example from the given timezones we care
+     * about for the US and its territories.
+     * Each of these will map to one of the
+     * ABBREVS_TO_LONGNAME zones (with variations
+     * for Daylight/Standard time as needed)
+     */
+    private static $US_TIMEZONE_IDS = [
+        "UTC",
+        "Pacific/Pago_Pago",
+        "Pacific/Saipan",
+        "America/Adak",
+        "America/Yakutat",
+        "America/Denver",
+        "America/North_Dakota/New_Salem",
+        "America/New_York",
+        "America/Los_Angeles",
+        "America/Phoenix",
+        "Pacific/Honolulu",
+    ];
+
+    /**
+     * For a given DateTimeZone instance, provide a label
+     * string that can be used in a user interface.
+     */
+    public static function getLabelForTimezone(\DateTimeZone $timezone)
+    {
+        $abbrev = (new \DateTime("now", $timezone))->format("T");
```

Comment:
Ooh, clever.

I wonder if this means on the day before DST ends, all the labels for the future will still reference their respective daylight savings time zone names even though DST will be ending. I don't think there's a general solution to that, though, because whether or not the DST name is correct will depend on when the user is trying to schedule something, not on, like, the natural order of the universe or something. Still, will be interesting to see. Anyway, clever!